src/cpp/src/visual_language/pipeline.cpp

```diff
@@ -382,40 +453,26 @@ class ov::genai::VLMPipeline::VLMPipelineImpl {
         const GenerationConfig& generation_config,
         const StreamerVariant& streamer
     ) {
-        ov::Tensor inputs_embeds;
+        // inputs_embeds, tokenized_input;
+        std::pair<ov::Tensor, ov::Tensor> processed_input;
         if (m_vlm_config.model_type == VLMModelType::MINICPM) {
-            inputs_embeds = get_inputs_embeds_minicpm(prompt, rgbs);
+            processed_input = get_inputs_embeds_minicpm(prompt, rgbs);
         } else if (m_vlm_config.model_type == VLMModelType::LLAVA) {
-            inputs_embeds = get_inputs_embeds_llava(prompt, rgbs);
+            processed_input = get_inputs_embeds_llava(prompt, rgbs);
         }
 
-        m_language.set_tensor("inputs_embeds", inputs_embeds);
-        size_t history_len = m_language.get_tensor("attention_mask").get_shape().at(1);
-        m_language.get_tensor("attention_mask").set_shape({1, history_len + inputs_embeds.get_shape()[1]});
-        std::fill_n(m_language.get_tensor("attention_mask").data<int64_t>(), m_language.get_tensor("attention_mask").get_size(), 1);
-        
-        m_language.get_tensor("position_ids").set_shape({1, inputs_embeds.get_shape().at(1)});
-        std::iota(m_language.get_tensor("position_ids").data<int64_t>(), m_language.get_tensor("position_ids").data<int64_t>() + m_language.get_tensor("position_ids").get_size(), history_len);
-        
-        m_language.get_tensor("beam_idx").set_shape({ BATCH_SIZE });
-        m_language.get_tensor("beam_idx").data<int32_t>()[0] = 0;
-
-        m_language.infer();
-
-        ov::Shape logits_shape = m_language.get_tensor("logits").get_shape();
-        auto attention_size = m_language.get_tensor("attention_mask").get_size();
-
-        int64_t sequence_len = m_language.get_tensor("logits").get_shape().at(1) - 1;
-        size_t vocab_size = m_language.get_tensor("logits").get_shape().back();
-        float* logits = m_language.get_tensor("logits").data<float>() + sequence_len * vocab_size;
-        int64_t out_token = std::max_element(logits, logits + vocab_size) - logits;
+        Sampler sampler = Sampler(m_tokenizer);
+        std::vector<SequenceGroup::Ptr> requests;
+        // request_id, input_ids, generation_config, block_size, enable_prefix_caching
```

Comment:
You can introduce variables with exactly same names and pass them to make_shared(). That way the compiler ensures that their names are valid.